File: 0704-binary-search/0704-binary-search.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
const search = function(nums, target) {
    let start =0,end = nums.length -1
    while(end >= start){
        const mid = Math.floor((end + start)/2)
        if(nums[mid] === target){
            return mid
        } else if (nums[mid] > target){
            end = mid-1
        } else{
            start = mid + 1
        }
    }
    return -1
};

File: 0704-binary-search/0704-binary-search.c
int search(int* nums, int numsSize, int target){
    int start = 0, end = numsSize -1;
    while (end >= start){
        int mid = (end + start) >> 1;
        if(nums[mid] == target){
            return mid;
        } else if (nums[mid] > target){
            end = mid - 1;
        } else {
            start = mid + 1;
        }
    }
    return -1;
}